reframe/core/containers.py

```diff
@@ -19,11 +19,31 @@ class ContainerPlatform(abc.ABC):
     #: :default: :class:`None`
     image = fields.TypedField(str, type(None))
 
-    #: The commands to be executed within the container.
+    #: The command to be executed within the container.
     #:
-    #: :type: :class:`list[str]`
-    #: :default: ``[]``
-    commands = fields.TypedField(typ.List[str])
+    #: If no command is given, then the default command of the corresponding
+    #: container image is going to be executed.
+    #:
+    #: ..versionchanged:: 3.4.2
+    #:   Changed the attribute name from `commands` to `command` and its type
+    #:   to a string.
+    #:
+    #: :type: :class:`str` or :class:`None`
+    #: :default: :class:`None`
+    command = fields.TypedField(str, type(None))
+
+    #: The pull command to be used to pull the container image.
+    #:
+    #: If an empty string is given as the pull command, then the default
+    #: pull command of the corresponding container platform is going to be
+    #: used to pull the image. If set to :class:`None`, then no pull action
+    #: is going to be performed by the container platform.
+    #:
+    #: ..versionadded:: 3.4.2
+    #:
+    #: :type: :class:`str` or :class:`None`
+    #: :default: ``''``
+    pull_command = fields.TypedField(str, type(None))
```

Comment:
Hmm, I'd say this is not relevant in the future - just as a workaround for my edge-case. 
At the end, you might generalise the pull by using skopeo from RH. They are able to transform in almost all formats.

Reply:
Maybe then have a boolean field indicating whether the image should be pulled or not. In any case you can perform custom pulling/loading using the `prerun_cmds`